MoviesApp/MoviesApp/Protocols+Presentors.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  Protocols.swift
+//  MoviesApp
+//
+//  Created by Dubrouski Nikita on 27.07.23.
+//
+
+import Foundation
+
+protocol MovieListView: AnyObject {
+    func showMovies(_ movies: [SearchModel], startIndex: Int)
+    func showError(_ error: Error)
+}
+
+
+protocol MovieListPresenter {
+    func viewDidLoad()
+    func loadMoreMovies()
+    func didSelectMovie(at index: Int)
+}
+
+protocol MovieListInteractor {
+    func getMovies(nameMovies: String, page: Int, completion: @escaping (Result<[SearchModel], Error>) -> Void)
+}
+
+
+class MovieListPresenterImpl: MovieListPresenter {
+    weak var view: MovieListView?
```

Comment:
final

Reply:
check modules names in my project, remove 

protocol -> MovieListPresenterProtocol
current class -> MovieListPresenter